core/templates/pages/contributor-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -47,12 +47,15 @@ export class ContributionOpportunitiesService {
 
   private _reloadOpportunitiesEventEmitter = new EventEmitter<void>();
   private _removeOpportunitiesEventEmitter = new EventEmitter<string[]>();
-  private _skillOpportunitiesCursor: string = null;
-  private _translationOpportunitiesCursor: string = null;
-  private _voiceoverOpportunitiesCursor: string = null;
-  private _moreSkillOpportunitiesAvailable = true;
-  private _moreTranslationOpportunitiesAvailable = true;
-  private _moreVoiceoverOpportunitiesAvailable = true;
+  // These properties are initialized using async methods
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
```

Comment:
What is the relevance of the async methods with the non-null assertion?

Reply:
`null` value for these properties doesn't portray anything. Properties are only initialized when these methods are called. Null here will have a same effect as undefined, but we'll have to add `| null` as a type and add certain checks everywhere the properties are used. So leaving them undefined doing non-null assertion.